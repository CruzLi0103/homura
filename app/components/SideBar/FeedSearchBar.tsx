import React from 'react';
import { SearchCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSearchBarProps = {
  link: string;
  onLinkChange: (link: string) => void;
  onSearch: (link: string) => void;
};

export const FeedSearchBar: React.FC<FeedSearchBarProps> = ({
  onSearch,
  link,
  onLinkChange,
}) => (
  <div className="flex bg-gray-300 text-gray-800">
    <div className="flex-1">
      <input
        value={link}
        onChange={(e) => onLinkChange(e.target.value)}
        className="w-full p-2 text-xs bg-transparent"
        placeholder="Feed URL"
      />
    </div>
    <IconContainer
      className="transition duration-300 text-gray-700 hover:bg-gray-700 hover:text-gray-300"
      onClick={() => onSearch(link)}
    >
      <SearchCircleFilledIcon />
    </IconContainer>
  </div>
);
