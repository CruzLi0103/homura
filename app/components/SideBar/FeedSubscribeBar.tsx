import React from 'react';
import { CheckCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSubscribeBarProps = {
  link: string;
  name: string;
  onNameChange: (name: string) => void;
  onConfirm: (name: string) => void;
};

export const FeedSubscribeBar: React.FC<FeedSubscribeBarProps> = ({
  link,
  name,
  onNameChange,
  onConfirm,
}) => (
  <>
    <div className="p-2 text-xs truncate">{link}</div>
    <div className="flex bg-gray-300 text-gray-800">
      <div className="flex-1">
        <input
          value={name}
          onChange={(e) => onNameChange(e.target.value)}
          className="w-full p-2 text-xs bg-transparent"
          placeholder="Save as Name"
        />
      </div>
      <IconContainer
        onClick={() => onConfirm(name)}
        className="transition duration-300 text-gray-700 hover:bg-gray-700 hover:text-gray-300"
      >
        <CheckCircleFilledIcon />
      </IconContainer>
    </div>
  </>
);
