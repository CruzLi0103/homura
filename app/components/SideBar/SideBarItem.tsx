import React from 'react';
import { RssIcon } from '../Icon';
import { LogoIcon } from '../LogoIcon';

export type SideBarItemProps = {
  name: string;
  count: number;
  active?: boolean;
  onClick?: () => void;
  url?: string;
  icon?: JSX.Element;
};

export const SideBarItem: React.FC<SideBarItemProps> = (props) => {
  return (
    <div
      className={`${
        props.active ? 'bg-gray-600' : ''
      } leading-8 text-gray-300 flex items-center justify-between cursor-pointer px-3`}
      onClick={props.onClick}
    >
      <div className="flex items-center overflow-x-hidden">
        <LogoIcon {...{ icon: <RssIcon />, ...props }} />
        <div className="text-xs ml-2 truncate">{props.name}</div>
      </div>
      {props.count > 0 && (
        <div className="text-xs text-gray-500">{props.count}</div>
      )}
    </div>
  );
};
