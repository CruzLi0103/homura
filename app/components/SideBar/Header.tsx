import React from 'react';
import { Mode } from '../../constants/Mode';
import {
  ArchiveIcon,
  MailFilledIcon,
  StarFilledIcon,
  ViewListIcon,
} from '../Icon';
import { SideBarItem } from './SideBarItem';

export enum OverviewTarget {
  All = 'All',
  Archive = 'Archive',
  Starred = 'Starred',
  Unread = 'Unread',
}

export type HeaderProps = {
  mode: Mode;
  count: number;
  active: OverviewTarget;
  onClick: (target: OverviewTarget) => void;
};

export const Header: React.FC<HeaderProps> = ({
  active,
  mode,
  count,
  onClick,
}) => {
  switch (mode) {
    case Mode.Starred:
      return (
        <SideBarItem
          active={active === OverviewTarget.Starred}
          icon={<StarFilledIcon />}
          name="Starred"
          count={count}
          onClick={() => onClick(OverviewTarget.Starred)}
        />
      );
    case Mode.Unread:
      return (
        <SideBarItem
          active={active === OverviewTarget.Unread}
          icon={<MailFilledIcon />}
          name="Unread"
          count={count}
          onClick={() => onClick(OverviewTarget.Unread)}
        />
      );
    default:
    case Mode.All:
      return (
        <>
          <SideBarItem
            active={active === OverviewTarget.All}
            icon={<ViewListIcon />}
            name="All Items"
            count={count}
            onClick={() => onClick(OverviewTarget.All)}
          />
          <SideBarItem
            active={active === OverviewTarget.Archive}
            icon={<ArchiveIcon />}
            name="Archive"
            count={0}
            onClick={() => onClick(OverviewTarget.Archive)}
          />
        </>
      );
  }
};
