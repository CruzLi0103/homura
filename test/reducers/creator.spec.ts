import { AnyAction } from 'redux';
import { Step } from '../../app/components/SideBar/BottomBar';
import {
  creatorReducer,
  reset,
  setLink,
  setLoading,
  setName,
  setParserError,
  stepToEnterName,
  stepToEnterUrl,
} from '../../app/features/creator/creatorSlice';

describe('reducers', () => {
  describe('creator', () => {
    it('should handle initial state', () => {
      expect(creatorReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle stepToEnterUrl', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: stepToEnterUrl }
        )
      ).toMatchSnapshot();
    });

    it('should handle stepToEnterName', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: stepToEnterName }
        )
      ).toMatchSnapshot();
    });

    it('should handle reset', () => {
      expect(
        creatorReducer(
          {
            step: Step.EnterUrl,
            link: 'link',
            name: 'name',
            loading: true,
            parseError: true,
          },
          { type: reset }
        )
      ).toMatchSnapshot();
    });

    it('should handle setLink', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: setLink, payload: 'link' }
        )
      ).toMatchSnapshot();
    });

    it('should handle setName', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: setName, payload: 'name' }
        )
      ).toMatchSnapshot();
    });

    it('should handle setParserError', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: setParserError, payload: true }
        )
      ).toMatchSnapshot();
    });

    it('should handle setLoading', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: setLoading, payload: true }
        )
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        creatorReducer(
          { step: null, link: '', name: '', loading: false, parseError: false },
          { type: 'unknown' }
        )
      ).toMatchSnapshot();
    });
  });
});
