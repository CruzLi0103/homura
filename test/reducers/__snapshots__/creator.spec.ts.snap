// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers creator should handle initial state 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": null,
}
`;

exports[`reducers creator should handle reset 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": null,
}
`;

exports[`reducers creator should handle setLink 1`] = `
Object {
  "link": "link",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": null,
}
`;

exports[`reducers creator should handle setName 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "name",
  "parseError": false,
  "step": null,
}
`;

exports[`reducers creator should handle stepToEnterName 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": 1,
}
`;

exports[`reducers creator should handle stepToEnterUrl 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": 0,
}
`;

exports[`reducers creator should handle unknown action type 1`] = `
Object {
  "link": "",
  "loading": false,
  "name": "",
  "parseError": false,
  "step": null,
}
`;
